### PR TITLE
Add Walla archive scraper

### PR DESCRIPTION
**Pull request by shaypal5**
## Summary
- add a Walla scraper based on monthly archive pages and pagination
- switch the checked-in Walla configs from RSS to scraper mode
- add fixture-backed tests for archive parsing, pagination, cutoff handling, and deduplication

## Validation
- `UV_CACHE_DIR=/tmp/uv-cache UV_TOOL_DIR=/tmp/uv-tools UV_PYTHON_INSTALL_DIR=/tmp/uv-python PYTHONPATH=src uv run pytest -q tests/integration/test_scrapers.py -k 'Walla or factory_helpers_create_expected_sources'`
- `UV_CACHE_DIR=/tmp/uv-cache UV_TOOL_DIR=/tmp/uv-tools UV_PYTHON_INSTALL_DIR=/tmp/uv-python PYTHONPATH=src uv run pytest -q tests/integration/test_pipeline.py tests/unit/test_pipeline_core.py`
- `uv run mypy src/denbust/sources/walla.py src/denbust/pipeline.py`
- `UV_CACHE_DIR=/tmp/uv-cache UV_TOOL_DIR=/tmp/uv-tools UV_PYTHON_INSTALL_DIR=/tmp/uv-python uv run ruff check src/denbust/sources/walla.py src/denbust/pipeline.py tests/integration/test_scrapers.py tests/integration/test_pipeline.py tests/unit/test_pipeline_core.py`

## Notes
- live sanity check: `WallaScraper(rate_limit_delay_seconds=0)` over 21 days recovered `https://news.walla.co.il/item/3823239`, the March 12 article that had already rolled off the RSS feed.

### CODE REVIEW

File: src/denbust/sources/walla.py [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+        date = self._parse_date(date_tag.get_text(" ", strip=True))
+        if not date:
+            return None
+
+        snippet_tag = article.find("p")
+        snippet = snippet_tag.get_text(" ", strip=True) if isinstance(snippet_tag, Tag) else ""
+
+        href = str(link.get("href", ""))
+        url = self._normalize_article_url(urljoin(WALLA_BASE_URL, href))
+
```
[the remark this answers is not in the document]
Comment (shaypal5) [reply src/denbust/sources/walla.py]: Implemented. `_parse_archive_item()` now validates the normalized URL before emitting an entry and only accepts `news.walla.co.il/item/...` article URLs. I also added tests for off-site/invalid links to keep that boundary explicit.
